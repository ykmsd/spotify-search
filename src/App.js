import React, { Component } from 'react';
import { connect } from 'react-redux';
import queryString from 'query-string';
import Search from '../src/components/Search/Search';
import {
  searchArtist,
  setToken,
} from './Actions/actionCreators';
import { config } from './config';

const clientId = config.CLIENT_ID;

class App extends Component {
  componentDidMount() {
    const parsedHash = queryString.parse(window.location.hash);
    if (!parsedHash.access_token) {
      const encodeURI = window.encodeURI(`https://accounts.spotify.com/authorize?client_id=${clientId}&redirect_uri=http://localhost:3000/callback/&response_type=token&state=123`); 
      window.location.href = encodeURI;
    } else {
      this.props.setToken(parsedHash.access_token);
    }
  }
  render() {
    return (
      <div className="App">
        <h1>Spotify Seach</h1>
        <Search {...this.props} />
      </div>
    );
  }
}

const mapStateToProps = state => state.reducer;

const mapDispatchToProps = dispatch => ({
  searchArtist: (searchTerm, token) => {
    dispatch(searchArtist(searchTerm, token));
  },
  setToken: (token) => {
    dispatch(setToken(token));
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
