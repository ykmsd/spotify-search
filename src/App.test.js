/* global expect, it, jest */
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { App } from './App';

const props = {
  setToken: jest.fn(),
  searchAlbum: jest.fn(),
  token: 'abc',
  items: [{ name: 'Taylor' }, { name: 'Gorillaz' }],
};

it('renders correctly', () => {
  const tree = renderer
    .create(<MemoryRouter><App {...props} /></MemoryRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
