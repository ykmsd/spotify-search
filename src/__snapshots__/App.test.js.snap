// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <div
    style={
      Object {
        "marginBottom": "20px",
      }
    }
  >
    <h1>
      Spotify Seach 
      <span
        aria-label="magnifying glass emoji"
        role="img"
      >
        ðŸ”Ž
      </span>
    </h1>
  </div>
  <div />
</div>
`;
